### PR TITLE
block: Add additional partial blocking templates

### PR DESCRIPTION
**Pull request by Amorymeltzer**
edit warring (ewpblock), account creation (acpblock), and emailing (epblock), plus a generic indef option (pblockindef)

Most of these made by or with consultation with @QEDK, but opened as draft since I think it'd be good to get further input on the language, timing, and other options.

### CODE REVIEW

**Comment by Amorymeltzer** [comment]
@QEDK You wanna take a quick look over this, in particular the default reason and summaries?  If you think they're fine, I think I'll just push this, then post on AN about some recent improvements and ask for feedback on options and language from a broader audience.

**Comment by QEDK** [comment]
Will look into this as soon as I get home tonight, hopefully in an hour or
so.

File: modules/twinkleblock.js [hunk 1]
```diff
@@ -1096,7 +1134,11 @@ Twinkle.block.blockGroupsPartial = [
 	{
 		label: 'Partial block reasons',
 		list: [
-			{ label: 'Partial block', value: 'uw-pblock', selected: true }
+			{ label: 'Generic partial block (custom reason)', value: 'uw-pblock', selected: true },
+			{ label: 'Generic partial block (custom reason) - indefinite', value: 'uw-pblockindef' },
+			{ label: 'Edit warring', value: 'uw-ewpblock', selected: true },
+			{ label: 'Misusing multiple aounts', value: 'uw-acpblock', selected: true },
```
Comment (QEDK) [inline modules/twinkleblock.js]: fix typo in label -> "misusing"

File: modules/twinkleblock.js [hunk 2]
```diff
@@ -984,13 +983,52 @@ Twinkle.block.blockPresetsInfo = {
...
+		reason: '[[WP:Edit warring|Edit warring]]',
+		summary: 'You have been [[WP:PB|blocked]] from editing certain areas of the encyclopedia to prevent further [[WP:DE|disruption]] caused by your engagement in an [[WP:EW|edit war]]'
+	},
+	'uw-acpblock': {
+		autoblock: true,
+		expiry: '24 hours',
+		nocreate: true,
+		pageParam: false,
+		reasonParam: true,
+		reason: 'Misuing [[WP:Sock puppetry|multiple accounts]]',
```
Comment (QEDK) [inline modules/twinkleblock.js]: typo in reason -> "Misusing"

Comment (QEDK) [reply modules/twinkleblock.js]: Shorten link to [[WP:SOCK]]

File: modules/twinkleblock.js [hunk 3]
```diff
@@ -984,13 +983,52 @@ Twinkle.block.blockPresetsInfo = {
...
+		summary: 'You have been indefinitely [[WP:PB|partially blocked]] from certain areas of the encyclopedia'
+	},
+	'uw-ewpblock': {
+		autoblock: true,
+		expiry: '24 hours',
+		nocreate: false,
+		pageParam: false,
+		reasonParam: true,
+		reason: '[[WP:Edit warring|Edit warring]]',
+		summary: 'You have been [[WP:PB|blocked]] from editing certain areas of the encyclopedia to prevent further [[WP:DE|disruption]] caused by your engagement in an [[WP:EW|edit war]]'
```
Comment (QEDK) [inline modules/twinkleblock.js]: Shorten to "You have been...disruption due to [[WP:EW|edit warring]]"

File: modules/twinkleblock.js [hunk 4]
```diff
@@ -984,13 +983,52 @@ Twinkle.block.blockPresetsInfo = {
...
 		summary: 'You have been [[WP:PB|partially blocked]] from certain areas of the encyclopedia'
+	},
+	'uw-pblockindef': {
+		autoblock: true,
+		expiry: 'infinity',
+		forRegisteredOnly: true,
+		nocreate: false,
+		pageParam: false,
+		reasonParam: true,
+		summary: 'You have been indefinitely [[WP:PB|partially blocked]] from certain areas of the encyclopedia'
```
Comment (QEDK) [inline modules/twinkleblock.js]: partially blocked -> blocked
+"editing" certain areas

File: modules/twinkleblock.js [hunk 5]
```diff
@@ -984,13 +983,52 @@ Twinkle.block.blockPresetsInfo = {
...
 	},
+
+	// Begin partial block templates, accessed in Twinkle.block.blockGroupsPartial
 	'uw-pblock': {
 		autoblock: true,
 		expiry: '24 hours',
 		nocreate: false,
 		pageParam: false,
 		reasonParam: true,
 		summary: 'You have been [[WP:PB|partially blocked]] from certain areas of the encyclopedia'
```
Comment (QEDK) [inline modules/twinkleblock.js]: partially blocked -> blocked
+"editing" certain areas

File: modules/twinkleblock.js [hunk 6]
```diff
@@ -984,13 +983,52 @@ Twinkle.block.blockPresetsInfo = {
...
+		expiry: '24 hours',
+		nocreate: true,
+		pageParam: false,
+		reasonParam: true,
+		reason: 'Misuing [[WP:Sock puppetry|multiple accounts]]',
+		summary: 'You have been [[WP:PB|blocked]] from creating accounts for misusing [[WP:Sock puppetry|multiple accounts]]'
+	},
+	'uw-epblock': {
+		autoblock: true,
+		expiry: '24 hours',
```
Comment (QEDK) [inline modules/twinkleblock.js]: set to infinity.

File: modules/twinkleblock.js [hunk 7]
```diff
@@ -984,13 +983,52 @@ Twinkle.block.blockPresetsInfo = {
...
+		nocreate: false,
+		pageParam: false,
+		reasonParam: true,
+		reason: '[[WP:Edit warring|Edit warring]]',
+		summary: 'You have been [[WP:PB|blocked]] from editing certain areas of the encyclopedia to prevent further [[WP:DE|disruption]] caused by your engagement in an [[WP:EW|edit war]]'
+	},
+	'uw-acpblock': {
+		autoblock: true,
+		expiry: '24 hours',
```
Comment (QEDK) [inline modules/twinkleblock.js]: set to infinity.

File: modules/twinkleblock.js (hunk 1 above)
Comment (Amorymeltzer) [reply modules/twinkleblock.js]: The other one yes, but not this one?

File: modules/twinkleblock.js (hunk 6 above)
Comment (Amorymeltzer) [reply modules/twinkleblock.js]: Should also get a `forRegisteredOnly: true` methinks

File: modules/twinkleblock.js (hunk 7 above)
Comment (Amorymeltzer) [reply modules/twinkleblock.js]: Couldn't this be applied to IPs as well @QEDK?  Probably the thing to do is create two presets, `forRegisteredOnly` and `forAnonOnly` with infinity and 48(?) hours as the defaults.

File: modules/twinkleblock.js (hunk 5 above)
Comment (Amorymeltzer) [reply modules/twinkleblock.js]: Here (and for the indef), I'm not sure about this: it's generic so someone could use it to apply non-editing PBs.  This is the edit summary for the template, so if the sysop didn't apply an editing block, they couldn't change the summary.

File: modules/twinkleblock.js (hunk 7 above)
Comment (QEDK) [reply modules/twinkleblock.js]: Yep, that's a better idea.

File: modules/twinkleblock.js (hunk 5 above)
Comment (QEDK) [reply modules/twinkleblock.js]: That's true, I would've guessed they can opt for the other presets but as a generic option, the current one fits best, true.

File: modules/twinkleblock.js (hunk 1 above)
Comment (QEDK) [reply modules/twinkleblock.js]: @Amorymeltzer "accounts" welp, seeing things it seems.

Comment (Amorymeltzer) [reply modules/twinkleblock.js]: Oy yes.  Common issue with me, involving a shortcut for er/contract-region.  Finally changed in my `.emacs`, thanks.

File: modules/twinkleblock.js (hunk 7 above)
Comment (Amorymeltzer) [reply modules/twinkleblock.js]: What's another template among friends, eh?  I suppose there's no need for the temporary one to be only anon-only.

**Comment by Amorymeltzer** [comment]
@QEDK should all be set now.  Could `Misusing multiple accounts` as the label for the `acpblock`s be too generic, inviting use for more than account-creation blocking?  This is all probably overkill, but maybe `Account creation block — misusing multiple accounts` and `Account creation block — misusing multiple accounts (indefinite)`?

**Comment by QEDK** [comment]
> @QEDK should all be set now. Could `Misusing multiple accounts` as the label for the `acpblock`s be too generic, inviting use for more than account-creation blocking? This is all probably overkill, but maybe `Account creation block — misusing multiple accounts` and `Account creation block — misusing multiple accounts (indefinite)`?

I doubt account creation blocks will be used frequently at all, but your call, imo more important stuff like DE/AE/3RR will also come up so more presets then, but mostly neutral feelings about this one.

**Pull request merged**